megamek/src/megamek/client/ui/SharedUtility.java

```diff
@@ -743,7 +744,12 @@ public static MovePath moveAero(MovePath md, Client client) {
                 if (!game.getBoard().contains(md.getLastStep().getPosition())) {
                     md.removeLastStep();
                     if(game.getOptions().booleanOption(OptionsConstants.ADVAERORULES_RETURN_FLYOVER)) {
-                        md.addStep(MoveStepType.RETURN);
+                        //Telemissiles shouldn't get a return option
+                        if (entity instanceof TeleMissile) {
```

Comment:
This does not appear to solve the problem, as I'm still able to fly missiles off the board and have them return.

I would suggest that you have the missile get removed from the game when it goes off board instead.

Reply:
Never mind, I needed to update the workspace.